### ISSUE TITLE
Plan for loading NIRSpec files for v0.3 (not a long-term solution)

### ISSUE THREAD

**Issue by hcferguson** [spacetelescope/mosviz#170]
This is to summarize just the near-term decisions made in the breakout 2/11 with @robelgeda @kassin @hcferguson and @drdavella. (@robelgeda -- there is one tweak relative to the whiteboard -- you can get the slit PA using Astropy as you suggested, see below). Also @SaOgaz might be helping with this work.

**Current status**

The current version of MOSViz was designed on the (incorrect) assumption that JWST pipeline outputs would be have separate directories would have the following structure:

- Root
  - Table to drive MOSViz 
  - 1D
    - One source per FITS file
  - 2D
    - One source per FITS file

The design also assumed (incorrectly) that enough metadata would be present in these FITS files to fill out the following fields in a table:

- ID (this is in the current files)
- Source RA & Dec
- Slit Width & Height (i.e. along the dispersion and cross dispersion, respectively)
- Slight Position Angle (East from North)

The slit was assumed to be centered on the source

**Plan for v0.3:**

For v0.3 we will be writing a stand-alone helper tool to extract the individual extensions from the _x1d and x2d files into the directory structure originally planned, naming each file with the SOURCE_ID. Then create the table and populate it from the following data available in the individual _x1d file extensions (one per source):

- ID = SOURCE_ID
- Source RA & Dec = SLIT_RA, SLIT_DEC
- Width = skycoord separation in  arcsec of short axis of REGION specification in the x1d file
- Height = skycoord separation in arcsec of long axis of REGION specification in the x1d file
- Position Angle = skycoord position_angle of the two region corners that run along the height direction

Note: Given the discussion on the Github issue﻿ #147, the current REGION specification is *NOT* the projection of the MSA aperture on the sky, and furthermore does not indicate which direction is along the dispersion and which is cross dispersion. So we know this is *WRONG* but it will give us something to work with until we work out how to get appropriate information into the _x1d and _x2d pipeline output files. 

This is the near-term work related to issue  #147. I will open a separate issue to begin to elaborate on the longer-term solution, which is rather complex and will need quite a few decisions that involve the NIRSpec, pipeline & data analysis teams before we can code anything in MOSViz.

**Comment by kassin** [comment]
THANK YOU!

**Comment by hcferguson** [comment]
@robel see #171....level 3 packaging is not what I thought it was!

**Comment by hcferguson** [comment]
@robel For now, let's concentrate on the level 2b loader, since the format seems to be close to what is described above. It seems like we may actually be in better shape than I thought for level 3, except for the lacking slit geometry information. 

I like your idea of leaving the level 2 data in the big FITS and just having the smarts to know that if the pathname has a [extension] it should go to the extension. That would allow us to leave the data alone on disk, and just take a (hopefully small)  performance hit to offset to the data in the FITS file. So if it's really just as easy to implement that as it is to build a helper tool to break up the files, go for it!

**Comment by robelgeda** [comment]
Closed by #182
